### PR TITLE
Added setting to remove playlist recommendations on the homepage #593

### PR DESCRIPTION
**Pull request by HelloIamarandomperson**
## Summary by CodeRabbit

- **New Features**
	- Added a setting to enable or disable the removal of playlists from the homepage.
	- Introduced a feature to skip the "Continue watching?" dialog when a video is paused.
	- Enhanced the configuration options for managing playlist length and YouTube Data API integration.
	- Added support for new localization features across multiple languages.

- **Bug Fixes**
	- Corrected JSON structure in various localization files to ensure proper formatting.

- **Chores**
	- Updated several dependencies in the project configuration.

### CODE REVIEW

**Comment by VampireChicken12** [comment]
@HelloIamarandomperson You don't have to manually update the locale files. Just revert the changes to the files other than en-US.json. Also, I noticed the feature folder is named "Tester".  Additionally, the feature name "playlistRemover" is unclear. It removes playlist recommendations from the YouTube home page.  "playlistRemover" is more fitting for a feature that removes playlists everywhere on YouTube. The setting key "enable_hide_playlist" could be clearer; perhaps "enable_hide_playlist_recommendations_from_home_page". I don't care how long setting keys are because users will rarely interact with them manually. 

Another thing that isn't a major issue is that you committed the code to the main branch instead of the dev branch. New features and bug fixes are committed to the dev branch; then, a pull request is made to the main branch when a new version is ready to be released.

**Comment by VampireChicken12** [comment]
https://github.com/YouTube-Enhancer/extension/compare/main...HelloIamarandomperson:extension:HidePlaylistOnHomepage#diff-c96f8cf28d48203fbd509ed71c0618dee10b6b08ed3bdde7a6be1d4a87f3e12dR5
4-5: Consider a more descriptive variable name.
"disableObserver" can be a bit unclear at first glance. Perhaps use "observerDisabled" or a similarly named boolean to clarify its meaning.

File: package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
     "globals": "^15.11.0",
     "i18next": "^23.7.3",
     "monaco-editor": "^0.51.0",
+    "npm": "^10.9.2",
```
[the remark this answers is not in the document]
Comment (VampireChicken12) [reply package.json]: @HelloIamarandomperson please apply this change

**Comment by VampireChicken12** [comment]
@coderabbitai read the files in the project and generate a class diagram using mermaid and a README in the markdown format

**Comment by HelloIamarandomperson** [comment]
> @HelloIamarandomperson You don't have to manually update the locale files. Just revert the changes to the files other than en-US.json. 
If I do that I get errors when trying to build the extension. It mentions its missing the keys, probably because it is.
Other than that, thanks, I've implemented all the changes. Not sure what I can do until I know how to fix that.

**Comment by VampireChicken12** [comment]
> > @HelloIamarandomperson You don't have to manually update the locale files. Just revert the changes to the files other than en-US.json.
> > If I do that I get errors when trying to build the extension. It mentions its missing the keys, probably because it is.
> > Other than that, thanks, I've implemented all the changes. Not sure what I can do until I know how to fix that.

Those errors only occur when building the extension for production, You can safely ignore them just use the dev build for now. Once the changes are merged I typically update the locale files from Crowdin

**Comment by HelloIamarandomperson** [comment]
I see two dev branches, upstream and orgin. I'm not sure which one I should be using here?

**Comment by VampireChicken12** [comment]
> I see two dev branches, upstream and orgin. I'm not sure which one I should be using here?

upstream

**Comment by VampireChicken12** [comment]
The property enable_hide_playlist_on_homepage might be renamed to enable_hide_playlist_recommendations_from_home_page

The feature folder name should match. Also the enable and disable function name should match.
@HelloIamarandomperson other than that I think the code is ready to be merged

**Pull request closed without merging**